sky/task.py

```diff
@@ -674,24 +674,30 @@ def get_preferred_store_type(self) -> storage_lib.StoreType:
         #  order:
         #  1. cloud decided in best_resources.
         #  2. cloud specified in the task resources.
-        #  3. the first enabled cloud.
+        #  3. if not specified or the specified cloud does not support storage,
+        #     use the first enabled storage cloud.
```

Comment:
```suggestion
        #  3. if neither of above strategies successfully determines a cloud,
        #     use the first enabled storage cloud.
```
nit: I was bit a confused on what you meant by specified, so just trying to elaborate.

Reply:
The two are not exactly the same, I rewarded it a bit. How does that look?
```
#  3. if not specified or the task's cloud does not support storage,
#     use the first enabled storage cloud.
```

Reply:
To unblock our release for 0.3.1, I am going to merge this PR for now. Please feel free to file another PR to fix the comment if feel the current one still confusing. : )